src/test/java/com/google/api/generator/engine/ast/ThisObjectValueTest.java

```diff
@@ -0,0 +1,66 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.api.generator.engine.ast;
+
+import static org.junit.Assert.assertThrows;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class ThisObjectValueTest {
+  @Test
+  public void testThisObjectValueTest() {
```

Comment:
Nit: Let's follow the existing `validFoobar_reasonOrEdgeCase` convention. In g3, folks are migrating away from having "test" as a suffix, since this is already a test class.

Reply:
Gotcha. I don't have change to make changes on g3 yet :( 👍